src/plugins/vis_augmenter/server/plugin.ts

```diff
@@ -22,16 +31,57 @@ export interface VisAugmenterPluginStart {}
 export class VisAugmenterPlugin
   implements Plugin<VisAugmenterPluginSetup, VisAugmenterPluginStart> {
   private readonly logger: Logger;
+  private readonly config$: Observable<VisAugmenterPluginConfigType>;
 
   constructor(initializerContext: PluginInitializerContext) {
     this.logger = initializerContext.logger.get();
+    this.config$ = initializerContext.config.create<VisAugmenterPluginConfigType>();
   }
 
-  public setup(core: CoreSetup) {
+  public async setup(core: CoreSetup) {
     this.logger.debug('VisAugmenter: Setup');
     core.savedObjects.registerType(augmentVisSavedObjectType);
     core.capabilities.registerProvider(capabilitiesProvider);
 
+    const config: VisAugmenterPluginConfigType = await this.config$.pipe(first()).toPromise();
+    const isAugmentationEnabled =
+      config.pluginAugmentationEnabled === undefined ? true : config.pluginAugmentationEnabled;
+
+    // Checks if the global yaml setting for enabling plugin augmentation is disabled.
+    // If it is disabled, remove the settings as we would not want to show these to the
+    // user due to it being disabled at the cluster level.
+    if (isAugmentationEnabled) {
```

Comment:
This is actually pretty neat that you are relying on the missing UI settings to disable it in the plugin. Calling it out in the comment here will be really helpful in the future though since it may not be obvious that you are doing it this way and its anon standard pattern in the repo.

Reply:
Will follow up in another pr with the tests